tests/kernel/workq/work_queue_api/src/main.c

```diff
@@ -640,13 +725,40 @@ static void ttriggered_work_cancel(const void *data)
 	/*work items not cancelled: triggered_work[1], triggered_work_sleepy*/
 }
 
-/*test cases*/
 /**
  * @brief Test work queue start before submit
+ * @details
+ * Test Objective:
+ * - Test the workqueue is initialized by defining the stack area used by
+ * its thread and then calling k_work_q_start().
  *
- * @ingroup kernel_workqueue_tests
+ * Test techniques:
+ * - Dynamic analysis and testing
+ * - Interface testing
+ *
+ * Prerequisite Conditions:
+ * - N/A
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Call the k_work_q_start() to create a work queue which has it's own
+ * thread that processes the work items in the queue.
+ *
```

Comment:
no check point here?

Reply:
I spilt the PR into 2 commits. The first commit use ztest to check this case and add code descriptions in the second commit.